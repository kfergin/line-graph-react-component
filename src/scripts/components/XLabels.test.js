import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import XLabels from './XLabels';

describe('<XLabels/>', () => {
	const props = {
		labels: [],
		width: 200,
		height: 100,
		shiftX: 10
	};
	it('should render its empty snapshot', () => {
		const tree = renderer.create(
			<XLabels {...props}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render its snapshot with string labels', () => {
		const labels = ['first', 'second', 'third'];
		const tree = renderer.create(
			<XLabels {...{...props, labels}}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render its snapshot with dots (instead of text)', () => {
		const labels = 4;
		const tree = renderer.create(
			<XLabels {...{...props, labels}}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	// is this the right test for the componentWillReceiveProps part?
	it('should set a new timestamp in state when labels change', () => {
		const labels = ['first', 'second', 'third'];
		const component = mount(
			<XLabels {...{...props, labels}}/>
		);
		const instance = component.instance();
		const spySetState = jest.spyOn(instance, 'setState');
		component.setProps({labels});
		expect(spySetState).not.toHaveBeenCalled();
		component.setProps({labels: 1});
		expect(spySetState).toHaveBeenCalled();
	});
});