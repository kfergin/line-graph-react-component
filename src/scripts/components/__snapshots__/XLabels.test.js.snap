// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XLabels/> should render its empty snapshot 1`] = `
<g
  className="x-labels"
  transform="translate(10)"
>
  <g
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</g>
`;

exports[`<XLabels/> should render its snapshot with dots (instead of text) 1`] = `
<g
  className="x-labels"
  transform="translate(10)"
>
  <g
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <circle
      cx={0}
      cy={98}
      fill="#ccc"
      r="0.75"
    />
    <circle
      cx={66.66666666666667}
      cy={98}
      fill="#ccc"
      r="0.75"
    />
    <circle
      cx={133.33333333333334}
      cy={98}
      fill="#ccc"
      r="0.75"
    />
    <circle
      cx={200}
      cy={98}
      fill="#ccc"
      r="0.75"
    />
  </g>
</g>
`;

exports[`<XLabels/> should render its snapshot with string labels 1`] = `
<g
  className="x-labels"
  transform="translate(10)"
>
  <g
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <text
      className="xLabel"
      x={0}
      y={100}
    >
      first
    </text>
    <text
      className="xLabel"
      x={66.66666666666667}
      y={100}
    >
      second
    </text>
    <text
      className="xLabel"
      x={133.33333333333334}
      y={100}
    >
      third
    </text>
  </g>
</g>
`;
